### PR TITLE
forest_fuzz: Instead of crashing between ops, crash in the middle of an op.

### PR DESCRIPTION
**Pull request by jamii**
Attempting to reproduce the bug I'm seeing in `jamii-compaction`, with no luck. But still a nice improvement to the fuzzer.

## Pre-merge checklist

Performance:

* [ ] Compare `zig benchmark` on linux before and after this pr.
    ``` sh
    # benchmark results before
    ...
    
    # benchmark results after
    ...
    ```
OR
* [x] I am very sure this PR could not affect performance.

### CODE REVIEW

**Comment by jamii** [comment]
The diff is not as bad as it looks, I just dedented Model.

File: src/lsm/forest_fuzz.zig [hunk 1]
```diff
@@ -535,28 +585,57 @@ pub fn generate_fuzz_ops(random: std.rand.Random, fuzz_op_count: usize) ![]const
...
+        // But this is difficult to test, so for now we'll avoid it.
+        const modifer_tag =
+            if (action == .compact and !action.compact.checkpoint)
+            fuzz.random_enum(
+                random,
+                FuzzOpModifierTag,
+                modifier_distribution,
+            )
+        else
+            FuzzOpModifierTag.normal;
```
Comment (sm2n) [inline src/lsm/forest_fuzz.zig]: The indentation on this if branch is weird, but I see that zig fmt thinks that's the right way to do it. I think it'd be more readable with a block even though it's a single expression. What do you think?

**Comment by sm2n** [comment]
I really like the trick with gating the crashes on ticks. On the whole this LGTM.

File: src/lsm/forest_fuzz.zig (hunk 1 above)
Comment (jamii) [reply src/lsm/forest_fuzz.zig]: Oh weird. I'll just push the if up onto the previous line.

**Review by sm2n — APPROVED** [review]

**Pull request merged**